utils/litemap/src/map.rs

```diff
@@ -210,6 +247,19 @@ impl<K: Ord, V> IndexMut<&'_ K> for LiteMap<K, V> {
         self.get_mut(key).expect("LiteMap could not find key")
     }
 }
+impl<K: Ord, V> FromIterator<(K, V)> for LiteMap<K, V> {
+    fn from_iter<I: IntoIterator<Item = (K, V)>>(iter: I) -> Self {
+        let iter = iter.into_iter();
+        let map = match iter.size_hint() {
+            (_, Some(upper)) => LiteMap::with_capacity(upper),
+            (lower, None) => LiteMap::with_capacity(lower),
+        };
+        match map.try_extend_from_sorted(iter) {
+            Ok(map) => map,
+            Err((map, rest)) => map.extend_from_unsorted(rest),
+        }
```

Comment:
Did you try something more along the lines of

```rust
for (key, value) in iter {
    match self.try_append(key, value) {
        Ok(_) => continue,
        Err((key, value)) => self.insert(key, value),
    }
}
```

Reply:
I agree, returning `(Self, Iter)` is a bit weird for Rust. My brain usually jumps to parsers that I've written in Haskell where you can't mutate anything, so you have to return the partial-success structure.

I actually changed the function to use `Option` in the same way that the `try_append()` function does. It should look cleaner now.

Reply:
Ping, did you see this optional suggestion?

Reply:
Oh I see. I originally thought this comment was obsolete once I changed the functions to return options. I didn't read it closely enough. 

I do like this more, even as far as performance. If a list was mostly sorted, but had one off-element in the middle, you'd still get optimal performance for all of the sorted sections, and we don't need the `extend_from` helper methods.